search/search_indexes.py

```diff
@@ -14,12 +14,75 @@
 
 from __future__ import unicode_literals
 
-from lxml import etree
+from collections import defaultdict
+from datetime import datetime, timedelta
+
+from django.conf import settings
 from haystack import indexes
+from lxml import etree
 
-from learningresources.models import LearningResource
+from learningresources.models import Course, LearningResource
 from taxonomy.models import Vocabulary
 
+INDEX_CACHE = {
+    "course": {},
+    "term": {},
+    "born": datetime.now(),
+}
+MAX_INDEX_AGE = timedelta(minutes=1)
+
+
+def reset_cache():
```

Comment:
`reset_cache` makes it sound like the function always resets the cache, is it possible to change the name to be clearer about its purpose? If I saw a `Cache().reset()` in the wild I would expect it to always throw everything away